web/src/stream_list.ts

```diff
@@ -901,15 +919,24 @@ export function set_event_handlers({
         const current_narrow_stream_id = narrow_state.stream_id();
         const current_topic = narrow_state.topic();
 
+        if (
+            user_settings.web_channel_default_view ===
+            web_channel_default_view_values.list_of_topics.code
+        ) {
+            browser_history.go_to_location(hash_util.by_channel_topic_list_url(stream_id));
+            return;
+        }
+
         if (current_narrow_stream_id === stream_id && current_topic !== undefined) {
             const channel_feed_url = hash_util.by_stream_url(stream_id);
             browser_history.go_to_location(channel_feed_url);
             return;
         }
 
         if (
-            user_settings.web_channel_default_view ===
-            web_channel_default_view_values.channel_feed.code
+            // We expect `on_stream_click` to handle other values of `web_channel_default_view`.
+            user_settings.web_channel_default_view !==
+            web_channel_default_view_values.top_topic_in_channel.code
```

Comment:
Is this change useful? It seems like you already added a handler above for `web_channel_default_view_values.list_of_topics.code`

Reply:
Removed this hunk, but we should also rename `on_stream_click`  in a follow-up to reflect it's a "go to channel feed" function now. Maybe `on_stream_click_show_channel_feed` would be fine